docs/modules/record-encryption/con-record-encryption-overview.adoc

```diff
@@ -73,7 +73,15 @@ It is recommended to use the KMS in a high availability (HA) configuration.
 Key rotation involves periodically replacing cryptographic keys with new ones and is considered a best practice in cryptography.
 
 The filter allows the rotation of Key Encryption Keys (KEKs) within the Key Management System (KMS). 
-When a KEK is rotated, the new key material is used for newly produced records. Existing records, encrypted with older KEK versions, remain decryptable as long as the previous KEK versions are still in the KMS.
+When a KEK is rotated, the new key material is eventually used for newly produced records. Existing records, encrypted with older KEK versions, remain decryptable as long as the previous KEK versions are still in the KMS.
+
+When the KEK is rotated in the external KMS, it will take up to 2 hours (by default) before all records produced by the filter
```

Comment:
Isn't it `encryptionDekRefreshAfterWriteSeconds` that matters for encryptions and this the default is 1 hour?

Reply:
It's tricky since a DEK can be used right up until expiry if there are no accesses for a long time. Keith had the example like:
- 01:00 - DEK A created 
- 02:00 - DEK A becomes eligible for refresh
- 02:59 - user accesses DEK A triggering async refresh but gets DEK A from cache

If a DEK is being used frequently it should refresh close to 1h.

Reply:
For the overview I don't know whether to get into the details or just talk about the upper bound

Reply:
I think the overview can assume topics with a regular flow of messages, as long as we document the need for message flow and how everything fits together in the extreme somewhere.